vignettes/potential-optimizers.Rmd

```diff
@@ -133,29 +133,26 @@ pre_mem <- function(n) {
 #### Proposed Optimized Code
 
 ```{r pre_allocation_code_op}
-pre_mem_opt <- function(n) {
+mem_alloc_opt <- function(n) {
   vec <- vector(length = n)
   for (i in seq_len(n)) {
     vec[i] <- i
   }
 }
-
 ```
 
 ### Benchmark
 
 ```{r pre_allocate_benchmark, echo = TRUE, warning = FALSE, message= FALSE}
-library("ggplot2")
-library("microbenchmark")
 n <- 100000
-autoplot(microbenchmark(pre_mem(n), pre_mem_opt(n), pre_mem_opt_unknown_type(n)))
+autoplot(microbenchmark(mem_alloc(n), mem_alloc_opt(n)))
 ```
 
 ## Vectorization
 
 ### Idea
 
-A golden rule in R programming is to access the underlying C/Fortran routines as quickly as possible; the fewer functions calls required to achieve this, the better. Many R functions are therefore vectorised, that is the function’s inputs and/or outputs naturally work with vectors, reducing the number of function calls required.
```

Comment:
Hello Sir, I wrote this based on my readings [from here](https://csgillespie.github.io/efficientR/programming.html#general). I realize that what I have written may seem wrong without context, so have a look at the above URL. Go to section 3.2 *General Advice* and in that you'll find the *golden rule* as mentioned by the author. Thanks.